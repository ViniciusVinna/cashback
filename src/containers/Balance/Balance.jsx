import React from 'react';
import PropTypes from 'prop-types';

import { BalanceStyled } from './Balance.styled';

const {
  BalanceWidget,
  Header,
  Title,
  Label,
  Value,
  Info,
  Item,
  ItemValue,
} = BalanceStyled;

const Balance = ({ balance, approved, pending }) => (
  <BalanceWidget data-testid="balance">
    <Header>
      <Title>Saldo</Title>
      <Value>{balance}</Value>
    </Header>

    <Info>
      <Item>
        <Label>Aprovado</Label>
        <ItemValue>{approved}</ItemValue>
      </Item>

      <Item>
        <Label>Pendente</Label>
        <ItemValue>{pending}</ItemValue>
      </Item>
    </Info>
  </BalanceWidget>
);

Balance.propTypes = {
  approved: PropTypes.string,
  balance: PropTypes.string,
  pending: PropTypes.string,
};

Balance.defaultProps = {
  approved: 'R$ 400,00',
  balance: 'R$ 245,00',
  pending: 'R$ 75,00',
};

export default Balance;
