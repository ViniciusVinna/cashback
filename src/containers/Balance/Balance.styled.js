import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const BalanceStyled = {
  BalanceWidget: styled('div')(({ theme }) => css`
    background-color: ${theme.color.n100};
    border-radius: ${theme.borderRadius.m};
    box-shadow: ${theme.boxShadow.modal};
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    margin: 0 ${theme.space.l};
    overflow: hidden;
    padding: ${theme.space.l};
    position: relative;
    text-align: left;
    z-index: ${theme.zIndex.default};
  `),
  Header: styled('header')`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 100%;
  `,
  Info: styled('div')`
    display: flex;
    width: 100%;
  `,
  Item: styled('div')(({ theme }) => css`
    display: flex;
    flex-direction: column;
    margin-right: ${theme.space.s};
    width: 100%;
  `),
  ItemValue: styled('div')(({ theme }) => css`
    font-size: ${theme.textSize.l};
    color: ${theme.color.o500};
    display: flex;
    width: 100%;
  `),
  Label: styled('label')(({ theme }) => css`
    color: ${theme.color.d400};
    font-size: ${theme.textSize.s};
    margin-bottom: ${theme.space.xxs};
    width: 100%;
  `),
  Title: styled('label')(({ theme }) => css`
    color: ${theme.color.d500};
    margin-bottom: ${theme.space.s};
  `),
  Value: styled('div')(({ theme }) => css`
    color: ${theme.color.black};
    font-size: ${theme.textSize.xxl};
    font-weight: 800;
    margin-bottom: ${theme.space.l};
  `),
};
