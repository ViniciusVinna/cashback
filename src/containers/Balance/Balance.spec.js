import React from 'react';

import Balance from './Balance';

describe('Balance', () => {
  it('it should render properly', () => {
    const { container } = render(
      <Balance
        balance="R$ 450,00"
        approved="R$ 72,00"
        pending="R$ 45,00"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
