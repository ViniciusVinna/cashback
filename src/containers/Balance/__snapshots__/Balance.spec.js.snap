// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance it should render properly 1`] = `
<div>
  <div
    class="css-1dx82la"
    data-testid="balance"
  >
    <header
      class="css-d7tq2t"
    >
      <label
        class="css-1miv8ke"
      >
        Saldo
      </label>
      <div
        class="css-111w3cr"
      >
        R$ 450,00
      </div>
    </header>
    <div
      class="css-1jy0m1q"
    >
      <div
        class="css-gi4i87"
      >
        <label
          class="css-nwvdww"
        >
          Aprovado
        </label>
        <div
          class="css-ybx47f"
        >
          R$ 72,00
        </div>
      </div>
      <div
        class="css-gi4i87"
      >
        <label
          class="css-nwvdww"
        >
          Pendente
        </label>
        <div
          class="css-ybx47f"
        >
          R$ 45,00
        </div>
      </div>
    </div>
  </div>
</div>
`;
